pyccl/halos/halo_model.py

```diff
@@ -318,44 +336,44 @@ def I_0_22(self, cosmo, k, a,
             cosmo (:class:`~pyccl.core.Cosmology`): a Cosmology object.
             k (float or array_like): comoving wavenumber in Mpc^-1.
             a (float): scale factor.
-            prof1 (:class:`~pyccl.halos.profiles.HaloProfile`): halo
+            prof (:class:`~pyccl.halos.profiles.HaloProfile`): halo
                 profile.
-            prof12_2pt (:class:`~pyccl.halos.profiles_2pt.Profile2pt`):
-                a profile covariance object returning the the
-                two-point moment of `prof1` and `prof2`.
             prof2 (:class:`~pyccl.halos.profiles.HaloProfile`): a
                 second halo profile. If `None`, `prof1` will be used as
                 `prof2`.
             prof3 (:class:`~pyccl.halos.profiles.HaloProfile`): a
                 second halo profile. If `None`, `prof1` will be used as
                 `prof3`.
-            prof34_2pt (:class:`~pyccl.halos.profiles_2pt.Profile2pt`):
-                a profile covariance object returning the the
-                two-point moment of `prof3` and `prof4`.
             prof4 (:class:`~pyccl.halos.profiles.HaloProfile`): a
                 second halo profile. If `None`, `prof3` will be used as
                 `prof4`.
+            prof12_2pt (:class:`~pyccl.halos.profiles_2pt.Profile2pt`):
+                a profile covariance object returning the the
+                two-point moment of `prof1` and `prof2`.
+            prof34_2pt (:class:`~pyccl.halos.profiles_2pt.Profile2pt`):
+                a profile covariance object returning the the
+                two-point moment of `prof3` and `prof4`.
 
         Returns:
              float or array_like: integral values evaluated at each
              value of `k`.
         """
         if prof3 is None:
-            prof3 = prof1
+            prof3 = prof
         if prof34_2pt is None:
             prof34_2pt = prof12_2pt
 
         self._get_ingredients(a, cosmo, False)
-        uk12 = prof12_2pt.fourier_2pt(prof1, cosmo, k, self._mass, a,
-                                      prof2=prof2, mass_def=self._mdef).T
-        uk34 = prof34_2pt.fourier_2pt(prof3, cosmo, k, self._mass, a,
-                                      prof2=prof4, mass_def=self._mdef).T
+        uk12 = prof12_2pt.fourier_2pt(cosmo, k, self._mass, a, prof,
+                                      prof2=prof2, mass_def=self.mass_def).T
+        uk34 = prof34_2pt.fourier_2pt(cosmo, k, self._mass, a, prof3,
+                                      prof2=prof4, mass_def=self.mass_def).T
         i04 = self._integrate_over_mf(uk12[None, :, :] * uk34[:, None, :])
         return i04
 
 
-def halomod_mean_profile_1pt(cosmo, hmc, k, a, prof,
-                             normprof=False):
+@warn_api()
+def halomod_mean_profile_1pt(cosmo, hmc, k, a, prof, *, normprof=None):
```

Comment:
Why `None` instead of `False`?

Reply:
Because `False` is a valid normalization.

I wanted to use something that is not `True` or `False` (because these are the only choices), as a placeholder in order not to break API. The decorator checks whether `normprof is None`, and if it is, it warns the user.
If we leave it empty, we break API. The default is intended to go completely in the next major version.

Reply:
Do **not** resolve this. I will revert it back to False, since it is not passed in the `kwargs` dictionary if the user doesn't explicitly declare it anyway. This will simplify things.

Reply:
Right, so I reverted the default back to `False` now, instead of `None`. The decorator will still throw a warning if `normprof` has not been specified. The only difference is that, in functions accepting `normprof` which call other functions accepting `normprof`, there won't be a double warning anymore (it's not needed anyway).

We can now resolve this, and the ones below.